crates/autopilot/src/run_loop.rs

```diff
@@ -234,7 +261,10 @@ impl RunLoop {
             }
 
             tracing::info!(url = %self.drivers[*index].url, "settling with solver");
-            match self.settle(id, &self.drivers[*index], solution).await {
+            match self
+                .settle(id, &self.drivers[*index], solution, &revealed)
+                .await
```

Comment:
The reason why you want to introduce `/reveal` instead of taking the calldata from `/settle` is because the request to `/settle` might fail or in some other way not return the calldata, right?

Reply:
Exactly. Autopilot could be killed while `settle` is executing and we also can't force driver to return early (before starting the submission process).
The only alternative way I can think of (if we want to avoid introducing `/reveal`) is to do it in some background autopilot worker where we would periodically check all auction ids with missing calldata and fetch them from drivers but it would require drivers to save that data in their own db which I don't like.

Reply:
Makes sense. 👍 
I'm very much looking forward to new smart contracts that make it so we no longer have to worry about internal buffer trading. 😩